kombu/connection.py

```diff
@@ -867,6 +867,8 @@ def _extract_failover_opts(self):
                 conn_opts['interval_step'] = transport_opts['interval_step']
             if 'interval_max' in transport_opts:
                 conn_opts['interval_max'] = transport_opts['interval_max']
+            if 'connect_total_timeout' in transport_opts:
```

Comment:
would love to see test coverage for the change

Reply:
Added two tests, for behavor with and without `connect_total_timeout`.

Reply:
Added test cases with and without the new option. I hope it's OK to use `time.sleep()` in tests.